### PR TITLE
Cohort subsets

### PR DESCRIPTION
**Pull request by tomwhite**
This addresses https://github.com/pystatgen/sgkit/issues/224#issuecomment-694350208 for specifying cohort subsets (for Garud H and PBS).

This is a draft since I haven't run it on MalariaGEN scale data yet, but I'd welcome any initial feedback.

### CODE REVIEW

**Comment by tomwhite** [comment]
I've now run this successfully on the MalariaGEN data:

H stats: https://nbviewer.jupyter.org/github/tomwhite/shiny-train/blob/sgkit/notebooks/gwss/sgkit_h12.ipynb
PBS: https://nbviewer.jupyter.org/github/tomwhite/shiny-train/blob/sgkit/notebooks/gwss/sgkit_pbs.ipynb

Example usage:

```
sg.Garud_h(ds, cohorts=["ao_col"]) # list of cohort IDs
```

```
sg.pbs(ds, cohorts=[("ao_col", "ga_gam", "gw")]) # list of cohort ID triples
```

File: sgkit/cohorts.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+def _cohorts_to_array(
+    cohorts: Sequence[Union[int, Tuple[int, ...], str, Tuple[str, ...]]],
+    index: Optional[pd.Index] = None,
+) -> ArrayLike:
+    """Convert cohorts or cohort tuples specified as a sequence of values or
+    tuples to an array of ints used to match samples in ``sample_cohorts``.
+
+    Cohorts can be specified by index (as used in ``sample_cohorts``), or a label, in
+    which case an ``index`` must be provided to find index locations for cohorts.
```
Comment (jeromekelleher) [inline sgkit/cohorts.py:24]: Any chance of a couple of simple examples here and the return values? I'm finding it a bit abstract and a concrete example would help understand what the function does.

**Review by jeromekelleher — APPROVED** [review]
This is powerful stuff, thanks @tomwhite! A minor comment above, merge away as you see fit.

File: sgkit/cohorts.py (hunk 1 above)
Comment (tomwhite) [reply sgkit/cohorts.py:24]: Thanks for the review @jeromekelleher. Added two examples in https://github.com/pystatgen/sgkit/pull/394/commits/f4b45ea3cba8e234c65c43944d135dc262fa5ec0

**Pull request merged**